dot/network/README.md

```diff
@@ -0,0 +1,178 @@
+#### `ping`
+
+This is a simple liveness check [protocol](https://docs.libp2p.io/concepts/protocols/#ping) that peers can use to
+quickly see if another peer is online - it is
+[included](https://github.com/libp2p/go-libp2p/tree/master/p2p/protocol/ping) with the official Go implementation of
+`libp2p`.
+
+#### `identify`
+
+The [`identify` protocol](https://docs.libp2p.io/concepts/protocols/#identify) allows peers to exchange information
+about each other, most notably their public keys and known network addresses; like [`ping`](#ping), it is
+[included with `go-libp2p`](https://github.com/libp2p/go-libp2p/tree/master/p2p/protocol/identify).
+
+#### Noise
+
+As described above, [Noise](http://noiseprotocol.org/) provides `libp2p` with its
+[key distribution](#identity--key-management) capabilities. The Noise protocol is
+[well documented](http://cryptowiki.net/index.php?title=Noise_Protocol_Framework) and the Go implementation is
+maintained [under the official](https://github.com/libp2p/go-libp2p-noise) `libp2p` GitHub organization. Noise defines a
+number of [variables](http://cryptowiki.net/index.php?title=Noise_Protocol_Framework#Noise_Variables) and
+[handshake patterns](http://cryptowiki.net/index.php?title=Noise_Protocol_Framework#Handshake_patterns) that
+participants in a peer-to-peer network can use to establish message-passing channels with one another.
+
+#### Yamux
+
+[Yamux (Yet another Multiplexer)](https://github.com/hashicorp/yamux) is a Golang library for
+[stream-oriented multiplexing](#stream-multiplexing) that is maintained by [HashiCorp](https://www.hashicorp.com/) - it
+implements a well defined [specification](https://github.com/hashicorp/yamux/blob/master/spec.md). Gossamer uses
+[the official `libp2p` adapter](https://github.com/libp2p/go-libp2p-yamux) for Yamux.
+
+#### Kademlia
+
+[Kademlia](https://en.wikipedia.org/wiki/Kademlia) is a battle-tested
+[distributed hash table (DHT)](https://en.wikipedia.org/wiki/Distributed_hash_table) that defines methods for managing a
+dynamic list of peers that is constantly updated in order to make a P2P network more resilient and resistant to attacks.
+Kademlia calculates a logical "distance" between any two nodes in the network by applying the xor operation to the IDs
+of those two peers. Although this "distance" is not correlated to the physical distance between the peers, it adheres to
+three properties that are [crucial to the analysis](https://en.wikipedia.org/wiki/Kademlia#Academic_significance) of
+Kademlia as a protocol - in particular, these three properties are:
+
+- the "distance" between a peer and itself is zero
+- the "distance" between two peers is the same regardless of the order in which the peers are considered (it is
+  [symmetric](https://en.wikipedia.org/wiki/Symmetry_in_mathematics))
+- the shortest "distance" between two peers does not include any intermediate peers (it follows the
+  [triangle inequality](https://en.wikipedia.org/wiki/Triangle_inequality))
```

Comment:
It might be worth mentioning that in Kademlia, peers save more information about peers that are closer to them. And because of this structure in peers, we can walk in the dht to the next closest peer to the peer that we are looking for.

There is a nice talk by feross, if you want to include such things https://youtu.be/kxHRATfvnlw?t=1029

Reply:
This is a little more detail than I'd like to get into in this document, but very interesting!